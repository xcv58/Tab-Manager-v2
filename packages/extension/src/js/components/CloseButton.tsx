import React from 'react'
import classNames from 'classnames'

export interface CloseButtonProps {
  /**
   * Is the button disabled
   */
  disabled?: boolean
  /**
   * The click handler
   */
  onClick: () => void
}

const CloseButton: React.FC<CloseButtonProps> = (props) => {
  const { onClick, disabled, ...restProps } = props
  return (
    <button
      {...restProps}
      {...{
        onClick,
        disabled,
      }}
      className={classNames(
        'inline-flex justify-center w-8 h-8 m-2 text-xl text-red-200 bg-transparent rounded-full disabled:opacity-75',
        {
          'hover:text-red-500 hover:bg-red-100 focus:outline-none focus:ring active:bg-red-300 active:text-red-700':
            !disabled,
          'cursor-not-allowed': disabled,
        }
      )}
    >
      x
    </button>
  )
}

export default CloseButton
