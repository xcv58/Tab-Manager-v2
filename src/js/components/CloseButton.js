import React from 'react'
import CloseIcon from '@material-ui/icons/Close'
import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'

const styles = theme => ({
  icon: {
    opacity: 0.6,
    '&:hover': {
      opacity: 1,
      color: theme.palette.error.main
    }
  }
})

@withStyles(styles)
export default class CloseButton extends React.Component {
  static defaultProps = {
    disabled: false
  }

  render () {
    const { classes, onClick, disabled } = this.props
    return (
      <IconButton {...{ onClick, disabled }} className={classes.icon}>
        <CloseIcon />
      </IconButton>
    )
  }
}
