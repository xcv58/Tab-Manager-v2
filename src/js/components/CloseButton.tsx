import React from 'react'
import CloseIcon from '@material-ui/icons/Close'
import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'

const styles = theme => ({
  icon: {
    opacity: 0.6,
    '&:hover': {
      opacity: 1,
      color: theme.palette.error.main
    }
  }
})

const CloseButton = props => {
  const { classes, onClick, disabled } = props
  return (
    <IconButton {...{ onClick, disabled }} className={classes.icon}>
      <CloseIcon />
    </IconButton>
  )
}

export default withStyles(styles)(CloseButton)
