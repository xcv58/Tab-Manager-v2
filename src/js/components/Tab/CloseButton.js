import React from 'react'
import { inject, observer } from 'mobx-react'
import CloseIcon from '@material-ui/icons/Close'
import IconButton from '@material-ui/core/IconButton'
import { withTheme } from '@material-ui/core/styles'

@withTheme()
@inject('dragStore')
@observer
export default class CloseButton extends React.Component {
  onClick = () => {
    const { removing, remove } = this.props.tab
    if (!removing) {
      remove()
    }
  }

  render () {
    const { theme } = this.props
    return (
      <IconButton
        onClick={this.onClick}
        style={{
          color: theme.palette.secondary.main
        }}
      >
        <CloseIcon />
      </IconButton>
    )
  }
}
