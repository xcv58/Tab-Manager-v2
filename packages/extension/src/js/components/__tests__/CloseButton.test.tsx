import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import CloseButton from 'components/CloseButton'

const props = {
  onClick: jest.fn(),
}

describe('CloseButton', () => {
  it('should render correct components', () => {
    render(<CloseButton {...props} />)
    expect(screen.getByRole('button')).toHaveTextContent('x')
    expect(screen.getByRole('button')).toBeEnabled()
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('should honor disabled', () => {
    render(<CloseButton {...props} disabled />)
    expect(screen.getByRole('button')).toBeDisabled()
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('should call onClick', () => {
    const onClick = jest.fn()
    render(<CloseButton onClick={onClick} />)
    fireEvent.click(screen.getByRole('button'))
    expect(onClick.mock.calls.length).toBe(1)
    fireEvent.click(screen.getByRole('button'))
    expect(onClick.mock.calls.length).toBe(2)
  })
})
