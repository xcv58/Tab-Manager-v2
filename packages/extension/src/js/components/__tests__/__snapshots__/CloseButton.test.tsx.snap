// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton should honor disabled 1`] = `
<button
  class="inline-flex items-center justify-center w-8 h-8 m-2 text-xl text-red-200 bg-transparent rounded-full disabled:opacity-75 cursor-not-allowed"
  disabled=""
>
  x
</button>
`;

exports[`CloseButton should render correct components 1`] = `
<button
  class="inline-flex items-center justify-center w-8 h-8 m-2 text-xl text-red-200 bg-transparent rounded-full disabled:opacity-75 hover:text-red-500 hover:bg-red-100 focus:outline-none focus:ring active:bg-red-300 active:text-red-700"
>
  x
</button>
`;
